test/functional/api/vim_spec.lua

```diff
@@ -81,6 +81,26 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_execute_lua', function()
```

Comment:
Yes, for instance for add_highlight I will probably recommend lua over call_atomic.  There should probably be a standard runtime lua function for "append some text to a buffer with certain highlights" instead of every rplugin maintaining an output buffer needs to reimplement it.